### PR TITLE
WIP: Support multiple repositories

### PR DESCRIPTION
**Pull request by briansloane**
This branch supports multiple repo paths in the repository configuration, delimited by a space.

ex: 
```
{"access_token": "<redacted>",
 "repository": "singer-io/tap-github singer-io/tap-pipedrive"}
```

TODO:

- [x] Loop through repos from config
- [x] add bookmarks for each repo + stream 
- [x] check for old and new bookmark format but only write new format
- [x] inject _sdc_repository with repo_path into the records

### CODE REVIEW

**Comment by briansloane** [comment]
I expect this PR will resolve https://github.com/singer-io/tap-github/issues/19

File: tap_github.py [hunk 1]
```diff
@@ -373,43 +365,47 @@ def do_sync(config, state, catalog):
...
-                        sub_stream = get_stream_from_catalog(sub_stream_id, catalog)
-                        stream_schemas[sub_stream_id] = sub_stream['schema']
-                        singer.write_schema(sub_stream_id, sub_stream['schema'],
-                                            sub_stream['key_properties'])
-
-                # sync stream and it's sub streams
-                state = sync_func(stream_schemas, config, state, mdata)
-
-            singer.write_state(state)
+    repositories = list(filter(None, config['repository'].split(' ')))
```
Comment (timvisher) [inline tap_github.py]: I think it would be better to rename the config key to `repositories` and have `repository` as an optional config that we could fall back to (and not split) if and only if `repositories` is not set.

Comment (briansloane) [reply tap_github.py]: Right. This misnomer pained me a little bit but less so than the idea of flat-out changing it and causing a breaking change for anybody who already had a working configuration. Your suggesting is appealing so that it is more properly named but I don't love the idea of having a config property around that isn't really needed.

Would it make sense to introduce `repositories` now, and then remove `repository` in the future when a major version bump is more justified for the tap?

Comment (timvisher) [reply tap_github.py]: > Would it make sense to introduce repositories now, and then remove repository in the future when a major version bump is more justified for the tap?

That is precisely the path forward that I would suggest. Support the old style (translating it to the new style on the fly with a log message) but prefer the new style. Then eventually we can major version bump it (definitely no need to do that now).

**Comment by briansloane** [comment]
I merged in the changes from @alisaraa to add `_sdc_repository` to the records and schemas. I also implemented bookmarking per repo+stream and support reading the older state format while only writing out to the new format.

**Comment by timvisher** [comment]
@nick-mccoy noticed that running this with an old style bookmark doesn't convert that bookmark to the new style. We should make sure that the translation is working.

**Comment by briansloane** [comment]
I just ran this branch locally and the state looks exactly like the format outlined in my previous comment. Here is the STATE message: 

```
{
  "value": {
    "bookmarks": {
      "singer-io/tap-adwords": {
        "commits": {
          "since": "2018-11-14T03:04:00.589269Z"
        },
        "comments": {
          "since": "2018-11-14T03:03:57.731198Z"
        }
      },
      "singer-io/tap-github": {
        "commits": {
          "since": "2018-11-14T03:03:57.458312Z"
        },
        "comments": {
          "since": "2018-11-14T03:03:54.738652Z"
        }
      }
    }
  },
  "type": "STATE"
}
```

Please let me know if you are seeing something otherwise.

**Comment by briansloane** [comment]
@nick-mccoy When I run the tap with that exact state I see it begin writing to the new format while leaving the old one in place.  This allows the tap to fail and then resume from an old state format if it hasn't fully transitioned everything to the new format. The tap looks for the new format first and falls back to the old one.

https://github.com/singer-io/tap-github/pull/33/files#diff-f26d495b1346c44875efc276d691e2fcR129

Note: It will only write the new state format if there are actual updates since the bookmark values.

**Comment by nick-mccoy** [comment]
@briansloane you're right, I was running a sync that didn't actually have any updates since the bookmark values.  I just ran again with older bookmark values, and the state updated as you described.  

tl;dr ignore me :)

**Comment by timvisher** [comment]
So I think it's important that the tap emits an updated state as the first thing that it does. That allows us to be aggressive about deprecating the old state format. So let's make that one of the first thing the tap does along that path.
